crates/ra_assists/src/handlers/change_lifetime_anon_to_named.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{AssistContext, AssistId, Assists};
+use ra_syntax::{ast, ast::TypeParamsOwner, AstNode, SyntaxKind};
+use std::collections::HashSet;
+
+// Assist: change_lifetime_anon_to_named
+//
+// Change an anonymous lifetime to a named lifetime.
+//
+// ```
+// impl Cursor<'_<|>> {
+//     fn node(self) -> &SyntaxNode {
+//         match self {
+//             Cursor::Replace(node) | Cursor::Before(node) => node,
+//         }
+//     }
+// }
+// ```
+// ->
+// ```
+// impl<'a> Cursor<'a> {
+//     fn node(self) -> &SyntaxNode {
+//         match self {
+//             Cursor::Replace(node) | Cursor::Before(node) => node,
+//         }
+//     }
+// }
+// ```
+// FIXME: How can we handle renaming any one of multiple anonymous lifetimes?
+pub(crate) fn change_lifetime_anon_to_named(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let lifetime_token = ctx.find_token_at_offset(SyntaxKind::LIFETIME)?;
+    let lifetime_arg = ast::LifetimeArg::cast(lifetime_token.parent())?;
+    if lifetime_arg.syntax().text() != "'_" {
+        return None;
+    }
+    let next_token = lifetime_token.next_token()?;
+    if next_token.kind() != SyntaxKind::R_ANGLE {
+        // only allow naming the last anonymous lifetime
+        return None;
+    }
+    match lifetime_arg.syntax().ancestors().find_map(ast::ImplDef::cast) {
```

Comment:
Maybe use this syntax, `let impl_def = lifetime_arg.syntax().ancestors().find_map(ast::ImplDef::cast)?;` to have less indentations

Reply:
thanks @bnjjj! good point - updated